tests/main/snap-download/task.yaml

```diff
@@ -0,0 +1,15 @@
+summary: Check that snap download works
+restore: |
+    rm -f *.snap
+execute: |
+    echo "Snap download can download snaps"
+    snap download hello-world
+    ls hello-world_*.snap
+
+    echo "Snap download understand --edge"
+    snap download --edge test-snapd-tools
+    ls test-snapd-tools_*.snap
+
+    echo "Snap download understands --devmode"
+    snap download --devmode --beta classic
```

Comment:
What does it mean to download with --devmode?  This isn't being installed. It feels like it would be okay to download nevertheless.